src/citrine/resources/design_workflow.py

```diff
@@ -43,15 +44,37 @@ def register(self, model: DesignWorkflow) -> DesignWorkflow:
             The newly created design workflow.
 
         """
-        # branch_id is in the body of design workflow endpoints, so it must be serialized.
-        # This means the collection branch_id might not match the workflow branch_id. The
-        # collection should win out, since the user is explicitly referencing the branch
-        # represented by this collection.
-        # To avoid modifying the parameter, and to ensure the only change is the branch_id, we
-        # deepcopy, modify, then register it.
-        model_copy = deepcopy(model)
-        model_copy.branch_id = self.branch_id
-        return super().register(model_copy)
+        # Importing locally to avoid circular dependency
+        from citrine.resources.branch import Branch, BranchCollection
+
+        if self.branch_id is None:
+            # There are a number of contexts in which hitting design workflow endpoints without a
+            # branch ID is valid, so only this particular usage is deprecated.
+            msg = ('Creating a design workflow without a branch is deprecated as of 1.19.0 and '
```

Comment:
When you merged main, this was pulled in. Given the deprecation warning, we don't need to allow users to create a design workflow without a branch anymore, although we could.